soc/xtensa/intel_adsp/common/irq-cavs.c

```diff
@@ -32,12 +32,16 @@ void z_soc_irq_enable(uint32_t irq)
 	case DT_IRQN(CAVS_INTC_NODE(1)):
 		dev_cavs = device_get_binding(DT_LABEL(CAVS_INTC_NODE(1)));
 		break;
+#if DT_NODE_EXISTS(DT_NODELABEL(cavs2))
```

Comment:
Please suggest if there a better way to make conditional definitions.

Reply:
Honestly?  Refactor the interrupt controller device so that it actually manages all the interrupts on the system and doesn't have to be instantiated once per core.   Where I've had the opportunity, I've done exactly that. :)

This is historical: early cavs versions in Zephyr were single core and managed a single register block, so the "obvious" (but, well, wrong) way to implement SMP was to just make more of them.  And so we pay in future complexity.

For now this is probably the best that can be done in a focused patch.  Though if you want to be clever there are macros defined that can enumerate over defined DT instances.  But rework and refactoring is definitely the current zeitgeist on these platforms, so...

Reply:
Oh, heh, except that I forget: this is cavs.  :)  The multitple definitions aren't per-core, they're per-level.  So... they should still be present in this device, no?  The "cavs2" devicetree node defines a controller that listens to the third of four external interrupts on the device.

Reply:
> Though if you want to be clever there are macros defined that can enumerate over defined DT instances. 

I think you could use these macros to generate a table and remove the switch block entirely. Have a look at `DT_INST_FOREACH_STATUS_OKAY`

Reply:
@andyross @MaureenHelm I dived into the spec a bit today and indeed the controller bits are always there, independently of the CPU cores, so it would seem easiest solution (and matching hw) is to just keep the the definitions although for the less-CPU-cores variant. This also avoids dropping all changes to this irq-cavs.c file. This is now reflected in the latest revision.